### PR TITLE
MonadFail migration

### PR DESCRIPTION
**Pull request by ejconlon**
Tests pass with Stackage LTS-15.11.

### CODE REVIEW

File: src/Text/Ginger/ErrInfo.hs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Text.Ginger.ErrInfo
+    ( ErrInfo (..)
+    ) where
+
+{-| A newtype around 'Either' that lets us define a useful 'MonadFail' instance. -}
```
Comment (tdammers) [inline src/Text/Ginger/ErrInfo.hs:7]: Style: prefer `--` over `{- -}`, especially for short comments.

**Comment by tdammers** [comment]
Hmm, I'm not 100% sure I like it like this.

Getting things to compile cleanly on 8.8 is of course necessary, but I think that rather than make `fail` work, we should get rid of it - it's kind of an ad-hoc solution to the problem of dealing with failures, and I'd much rather resort to a more "proper" solution, probably something along the lines of a more descriptive error type and `ExceptT`.

I'm particularly unhappy with extending `runGinger`, which is supposed to be (technically and morally) pure to return an arbitrary `m`.

I'll do some thinking on this and get back to you.

**Comment by tdammers** [comment]
On this note: #49

**Comment by ejconlon** [comment]
Ok, thanks for looking!

**Pull request closed without merging**